### PR TITLE
Enqueue final print jobs for a UserContest & its dependents when its `frozen` bit changes

### PR DESCRIPTION
**Pull request by raisfathin**
Depends on #46.

Full print worker UI will be implemented later.

This also invalidates final print jobs for a UserContest (but not its dependents!). 

The merged PDFs are available under media. This is already available in dev env with docker as per https://github.com/ioi-2022/translation/blob/master/nginx/conf.d/ioi.conf#L15.

Some TODOs still remain:
1. Should we create fresh PDF like we do for draft print jobs to ensure we don't change PDFs attached to (existing) final print jobs?
2. What to do with dependent UserContests when one gets thawed? Should be trivial if we decide to generate fresh PDFs for (1)
3. Should we invalidate finished jobs?

### CODE REVIEW

**Comment by fushar** [comment]
> Should we create fresh PDF like we do for draft print jobs to ensure we don't change PDFs attached to (existing) final print jobs?

Maybe no need to do this. This way merged PDFs for a country will be always the same for all countries which require them.

> What to do with dependent UserContests when one gets thawed? Should be trivial if we decide to generate fresh PDFs for (1)

By not generating fresh PDFs, dependent UserContests will automatically get the latest (most recently updated) merged PDFs. 

> Should we invalidate finished jobs?

Don't we have `DONE` job state for this?

**Comment by raisfathin** [comment]
>> Should we create fresh PDF like we do for draft print jobs to ensure we don't change PDFs attached to (existing) final print jobs?

> Maybe no need to do this. This way merged PDFs for a country will be always the same for all countries which require them.

>> What to do with dependent UserContests when one gets thawed? Should be trivial if we decide to generate fresh PDFs for (1)

> By not generating fresh PDFs, dependent UserContests will automatically get the latest (most recently updated) merged PDFs.

Is there any concern with printed documents not matching the document linked in the worker page (due to updated translation)?

>> Should we invalidate finished jobs?

> Don't we have DONE job state for this?

Right, I think we decided to leave finished jobs behind. I suppose my question is whether we should invalidate those that haven't been printed. If we don't generate fresh names for merged PDF and keep updating the file, the pending jobs will automatically use the updated translation later.

**Comment by fushar** [comment]
@raisfathin Updating a translation after it's been finalized should be a very very rare case which can be handled manually on case-by-case basis. I think the current solution should suffice for our purposes.

File: trans/utils/print_job_queue.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                                        user_contest.user.num_of_contestants):
+        return
+
+    logger.info('enqueueing {}', file_paths_with_counts)
+    user_contest.final_print_job = queue.enqueue_final_print_job(
+        file_paths_with_counts, user_contest.user)
+    user_contest.save()
+
+
+def _queue_dependent_final_print_job_if_completed(user_contest):
```
Comment (fushar) [inline trans/utils/print_job_queue.py:93]:
```suggestion
def _queue_dependent_final_print_jobs_if_completed(user_contest):
```

**Review by fushar — APPROVED** [review]

File: trans/utils/print_job_queue.py (hunk 1 above)
Comment (raisfathin) [reply trans/utils/print_job_queue.py]: Done.

**Pull request merged**